src/usdt.cpp

```diff
@@ -75,16 +96,15 @@ usdt_probe_list USDTHelper::probes_for_pid(int pid)
 {
   read_probes_for_pid(pid);
 
-  std::string path = bpftrace::get_pid_exe(pid);
-  if (usdt_provider_cache.find(path) == usdt_provider_cache.end())
-    path = bpftrace::path_for_pid_mountns(pid, path);
-
   usdt_probe_list probes;
-  for (auto const &usdt_probes : usdt_provider_cache[path])
+  for (auto const &path : usdt_pid_to_paths_cache[pid])
```

Comment:
So we cache pid → path here, and then the existing logic of the path caching is used right?

Reply:
Yes, exactly